### PR TITLE
Warm-up dependency on number of connections

### PR DESCRIPTION
**Pull request by orshalev**
Signed-off-by: Or Shalev <orsh@mellanox.com>

### CODE REVIEW

File: src/Client.cpp [hunk 1]
```diff
@@ -578,6 +576,13 @@ ::initBeforeLoop()
 
 					if (!g_pApp->m_const_params.pPlaybackVector) {
 						struct itimerval timer;
+						g_warmup_usec = 50000 /*first connection's first packet ttl*/ +
+								connections * 100 /*other connections' first packet ttl*/;
```
Comment (igor-ivanov) [inline src/Client.cpp]: Please avoid using comment inside expressions.

File: src/Client.cpp [hunk 2]
```diff
@@ -34,6 +34,7 @@
 #include "Switches.h"
 
 TicksTime s_startTime, s_endTime;
+uint32_t g_warmup_usec;
```
Comment (igor-ivanov) [inline src/Client.cpp]: What is a reason to use usec instead of msec for this value. I see in code that in all places you use de facto msec.

Comment (igor-ivanov) [reply src/Client.cpp]: please move this variable to `class App` and use `g_pApp->`

**Review by igor-ivanov — COMMENTED** [review]
please modify commit according feedback

File: src/Client.cpp (hunk 2 above)
Comment (orshalev) [reply src/Client.cpp]: struct itimerval's units are in usec:
timer->it_interval.tv_usec
But Sockperf's constants are in millisec.

**Comment by orshalev** [comment]
Please verify it:
Run a sockperf test with a large number of connections and check that it sets a higher warmup than 400msec (6000 connections should produce a 650 msec warmup).
This will be printed in the sockperf summary.
@liranoz12

**Comment by igor-ivanov** [comment]
merged (see https://github.com/Mellanox/sockperf/commit/c351dbfe7)

**Pull request closed without merging**